### ISSUE TITLE
rule object-shorthand "never" should not complain about destructuring assignment

### ISSUE THREAD

**Issue by atte-kemppila** [eslint/eslint#5488]
**What version of ESLint are you using?**
v2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Espree

**What did you do? Please include the actual source code causing the issue.**

```
/* eslint-env es6 */
/* eslint object-shorthand: [2, "never"] */

// object initializer

let a = 1, b = 2;

let o = {
    a: a,
    b,                  // object-shorthand error, as expected
    fb: function() {},
    fa() {}             // object-shorthand error, as expected
}


// destructuring assignment

let {a, b} = o;  // object-shorthand error, not expected

let arr = [{a: 1, b: 1}, {a: 2, b: 2}];

for (let {a, b} of arr) {}  // object-shorthand error, not expected
```

**What actually happened? Please include the actual, raw output from ESLint.**
```
  10:5   error  Expected longform property syntax  object-shorthand
  12:5   error  Expected longform method syntax    object-shorthand
  18:6   error  Expected longform property syntax  object-shorthand
  18:9   error  Expected longform property syntax  object-shorthand
  22:11  error  Expected longform property syntax  object-shorthand
  22:14  error  Expected longform property syntax  object-shorthand

✖ 6 problems (6 errors, 0 warnings)
```

**What did you expect to happen?**
Rule object-shorthand "never" should not complain about destructuring assignment. Lines 18 and 22 in the example should not produce error.

**Comment by alberto** [comment]
Maybe I was too quick to label it as a bug, since I just realized you can also use the long form for destructuring too:

```
let {a: a, b: b} = o;

for (let {a:a, b:b} of arr) {}
```

Not sure if this should be an option, though.

**Comment by michaelficarra** [comment]
So... there's no way to get the old behaviour? This probably should have been left in as an option.

**Comment by alberto** [comment]
Sorry, it's my bad this got merged before we agreed on a decision. :/

@eslint/eslint-team should this rule deal with shorthand in destructuring assignments or is this rule only concerned about object literals?

**Comment by ilyavolodin** [comment]
Going to reopen this issue. Sorry, my bad for not double-checking issue before merging code change (I actually did a few times and didn't merge the code, but this time around, I forgot).

**Comment by BYK** [comment]
:+1: to @michaelficarra

**Comment by mysticatea** [comment]
I read [the doc](http://eslint.org/docs/rules/object-shorthand), `object-shorthand` was a rule about object literals (object initializers). So I think making the rule covering destructuring assignments is an enhancement.

Though I don't have any objection,  I'd like to separate it to a new issue.

**Comment by nzakas** [comment]
Agree with @mysticatea. This rule was checking destructuring by default and it should not have. I'm unsure about an option vs. a new rule, but agree that should be a separate issue.

**Comment by alberto** [comment]
Closing then, since this was indeed a bug.
